client/cherami/outputhostconnection.go

```diff
@@ -175,7 +173,23 @@ func (conn *outputHostConnection) isClosed() bool {
 	return atomic.LoadInt32(&conn.closed) != 0
 }
 
+// drainReadPipe reads and discards all messages on
+// the outputHostStream until it encounters
+// a read stream error
+func (conn *outputHostConnection) drainReadPipe() {
+	for {
+		if _, err := conn.outputHostStream.Read(); err != nil {
+			return
+		}
```

Comment:
here should we try to write to the deliveryCh in a non-blocking way? That way, we can do a best-effort in delivering some messages if the application is still processing. That way we won't just drop the tail in case, say, the server is being restarted.